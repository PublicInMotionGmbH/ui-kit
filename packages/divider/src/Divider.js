import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'

export const moduleName = 'divider'

const propTypes = {
  /** Additional class name. */
  className: PropTypes.string,

  /** Changes top and bottom padding of divider to smaller. */
  small: PropTypes.bool
}

const defaultProps = {
  small: false
}

/**
 * Component which represents Divider.
 *
 * @param {object} props
 * @param {string} [props.className]
 * @param {boolean} [props.small]
 * @returns {React.Element}
 */
function Divider (props) {
  const { className, children, small, ...passedProps } = props
  const dividerCls = buildClassName(moduleName, className, { small })
  const spanCls = buildClassName([moduleName, 'text'])

  return (
    <div className={dividerCls} {...passedProps}>
      { children && <span className={spanCls}>{children}</span> }
    </div>
  )
}

Divider.displayName = 'Divider'

Divider.propTypes = propTypes
Divider.defaultProps = defaultProps

export default Divider
