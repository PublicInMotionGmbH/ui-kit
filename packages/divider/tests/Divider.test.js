import React from 'react'
import { shallow } from 'enzyme'

import Divider from '../src/Divider'

describe('<Divider />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Divider />)

    expect(wrapper).toMatchSnapshot()
  })
  it('renders children correctly', () => {
    const wrapper = shallow(<Divider>Test divider</Divider>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render children inside component', () => {
    const insideText = 'Test text'
    const wrapper = shallow(<Divider>{insideText}</Divider>)

    expect(wrapper.text()).toBe(insideText)
  })
})
