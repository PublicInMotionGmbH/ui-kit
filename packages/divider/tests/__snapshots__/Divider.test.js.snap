// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> renders children correctly 1`] = `
<div
  className="talixo-divider"
>
  <span
    className="talixo-divider__text"
  >
    Test divider
  </span>
</div>
`;

exports[`<Divider /> renders correctly 1`] = `
<div
  className="talixo-divider"
/>
`;
